### PR TITLE
Perl algorithm provider for the processing plugin

### PR DESCRIPTION
**Pull request by ajolma**
This is an initial addition to the processing plugin to allow running Perl scripts. It works in the sense that it can be added to the processing tools and a hello world script can be run from the Create new Perl script dialog. I've written this by copying the R script extension so it still has some things from that. There is something wrong with the icons although I added a basic Perl icon. I guess eventually this would need a Perl section in the on-line processing tools repository. I would initially add one Perl processing tool there from my GDALToolsWithPerl repo. I'm not sure what would be the best time to merge this to the main repository but it could be merged now at least technically. I'm not using Windows so I would have only very limited chances on working on that part.

### CODE REVIEW

File: python/plugins/processing/algs/perl/PerlUtils.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                PerlUtils.getConsoleOutputFilename()
+            ]
+
+        else:
+            os.chmod(PerlUtils.getPerlScriptFilename(), stat.S_IEXEC | stat.S_IREAD
+                     | stat.S_IWRITE)
+            command = 'PERL CMD BATCH --vanilla ' + PerlUtils.getPerlScriptFilename() \
+                + ' ' + PerlUtils.getConsoleOutputFilename()
+
+        proc = subprocess.Popen(
```
Comment (luipir) [inline python/plugins/processing/algs/perl/PerlUtils.py]: For new code, make sense to use QgsTask? what do you think @nyalldawson @volaya @alexbruy ?

**Comment by alexbruy** [comment]
We are now trying to reduce number of Processing providers shipped by default. IMHO better to publish it as a separate provider plugin at the QGIS Official Plugins repository

**Comment by ajolma** [comment]
Do you mean as a plugin that adds to the processing plugin or a completely separate plugin? I think that from a user's point of view it might be better to have a central place for processing tools & unified look and feel. Are there any examples of plugins that change the processing plugin - right now I had to make some changes to the core & gui of the processing plugin itself and haven't yet looked how difficult it is to attach to it from outside.

**Comment by alexbruy** [comment]
> Do you mean as a plugin that adds to the processing plugin or a completely separate plugin?

Sorry, I don't understand your question. IMHO it should be separate Processing provider plugin. You can find example in the QGIS code [processing/algs/exampleprovider](https://github.com/qgis/QGIS/tree/master/python/plugins/processing/algs/exampleprovider) or look at existing provider plugins in the QGIS plugins repository (search by processing tag)

**Comment by volaya** [comment]
Nice work. As Alex said, much better to have it as a separate plugin that extends the Processing toolbox when installed, since that's what we are trying to do with all providers. Let us know if you need help with that

**Comment by ajolma** [comment]
Thanks for the comments. I'll look into extending the processing toolbox from another plugin.

**Comment by ajolma** [comment]
This is now here: https://plugins.qgis.org/plugins/perlprocessing/

**Pull request closed without merging**